File: lua/mlgsim.lua
doritosCost = 30
doritosCount = 0
doritosPtsTurn = 10
dewPtsTurn = 20
dewCost = 60
dewCount = 0

printf = function(s,...)
           return print(s:format(...))
         end --function declaration of printf
function CalcPtsTurn()
  local dor = doritosCount*doritosPtsTurn
  local dew = dewCount*dewPtsTurn
  return dew+dor
end --CalcPtsTurn function
function main()
  mlgpoints = 100;
  print("\n\t-----MLG Simulator by Alex Pawelko-----")
  for i=1,15,1
  do
    printf("\n\tYou are on turn %i.",i)
    if(i > 1) then
      printf("\n\tYou earned %i MLG Points this turn!",CalcPtsTurn())
    end
    mlgpoints = mlgpoints + CalcPtsTurn()
    printf("\n\tYou have %i MLG Points.",mlgpoints)
    printf("\n\tCurrently you have %i Doritos and %d litre(s) of Mountain Dew.",doritosCount,dewCount)
    
    printf("\n\tDo you want to buy any Doritos? Doritos cost %i and earn %d MLG Points per turn. If you do not want to buy any type 0.",doritosCost,doritosPtsTurn)
    amount = io.read("*number")
    if(amount == nil) then
      print("You fail, that's not an integer!")
      return nil;
    elseif amount % 1 ~= 0 then
      print("You fail, that's not an integer!")
      return nil;
    end --if statement
    if amount > 0 then
      for attempt = 0,amount-1,1 do
        if ((mlgpoints - doritosCost) >= 0) then
          doritosCount = doritosCount + 1
          mlgpoints = mlgpoints - doritosCost
        else 
          print("You didn't have enough MLG Points to buy all those Doritos!")
          break
        end --if statement  
      end --for loop
      printf("\n\tYou now have %i Dorito(s)!",doritosCount)
    elseif amount < 0 then
      print("\n\tMATH DOESN'T WORK THAT WAY!")
    end --if statement
    
    
    printf("\n\tDo you want to buy Mountain Dew? Mountain Dew costs %i per litre and earns %d MLG Points per turn. If you do not want to buy any type 0.",dewCost,dewPtsTurn)
    amount = io.read("*number")
    if(amount == nil) then
      print("You fail, that's not an integer!")
      return nil;
    elseif amount % 1 ~= 0 then
      print("You fail, that's not an integer!")
      return nil;
    end --if statement
    if amount > 0 then
      for attempt = 0,amount-1,1 do
        if mlgpoints - dewCost >= 0 then
          dewCount = dewCount + 1
          mlgpoints = mlgpoints - dewCost
        else 
          print("You didn't have enough MLG Points to buy all that Mountain Dew!")
          break
        end --if statement  
      end --for loop
      printf("\n\tYou now have %i litre(s) of Mountain Dew!",dewCount)
    elseif amount < 0 then
      print("\n\tMATH DOESN'T WORK THAT WAY!")
    end --if statement
    printf("\n\tTurn %i has ended.",i)
    print("\n\t------------------------------------------------");
  end --for loop
end --main function
main();

File: c/mlgsim.c
#include "mlgsim.h"


static struct Doritos {
    int PtsTurn;
    int cost;
    int count;
};

static struct Dew {
    int PtsTurn;
    int cost;
    int count;
};

struct Doritos doritos;
struct Dew dew;

void assign(void) {
    doritos.cost = 30;
    doritos.PtsTurn = 10;
    dew.cost = 60;
    dew.PtsTurn = 20;
}

int main(int argc, char *argv[]) {
    assign();

    int MLGPoints = 100; //default value

    printf("\n\t------MLG Simulator by Alex Pawelko-----\t\n");

    for(int i = 1; i < 16; i++) {
        printf("\n\tYou are on turn %i.", i);
        printf("\n\tYou earned %d MLG Points this turn!",CalcPtsTurn());
        MLGPoints += CalcPtsTurn();
        printf("\n\tYou have %i MLG Points.", MLGPoints);
        printf("\n\tCurrently you have %i Doritos and %d litre(s) of Mountain Dew.", doritos.count,dew.count);

        printf("\n\tDo you want to buy any Doritos? Doritos cost %i and earn %d MLG Points per turn. If you do not "
                       "wish to buy any type 0.\n",doritos.cost, doritos.PtsTurn);
        int amount;

        scanf("%d",&amount);
        if (amount > 0) {
            int attempt = 0;
            for(; attempt < amount; attempt++) {
                if(MLGPoints - doritos.cost >= 0) {
                    ++doritos.count;
                    MLGPoints -= doritos.cost;
                } else {
                    printf("\n\tYou didn't have enough MLGPoints to buy all those Doritos.");
                    break;
                }
            }
            printf("\n\tYou now have %i Doritos.", doritos.count);
        } else if (amount < 0) {
            printf("\n\tDid you fail Grade 2 math? You can't buy a negative amount of Doritos.");
        }

        printf("\n\tDo you want to buy any Mountain Dew? Mountain Dew costs %i per litre and earn %d MLG Points per "
                       "turn. If you do not wish to buy any type 0.\n",dew.cost, dew.PtsTurn);

        int amount1;
        scanf("%d",&amount1);
        if(amount1 > 0) {
            int attempt = 0;
            for(; attempt < amount1; attempt++) {
                if(MLGPoints - dew.cost >= 0) {
                    ++dew.count;
                    MLGPoints -= dew.cost;
                } else {
                    printf("\n\tYou didn't have enough MLGPoints to buy all that Mountain Dew.");
                    break;
                }
            }
            printf("\n\tYou now have %i litres of Mountain Dew.",  dew.count);
        } else if(amount1 < 0) {
            printf("\n\tDid you fail Grade 2 math? You can't buy a negative quantity of Mountain Dew.");
        }
        printf("\n\tTurn %i has ended.", i);
        printf("\n\t------------------------------------------------");
    }

    printf("The game has ended with you scoring %i MLG Points, %d Doritos and %d litres of Mountain Dew!",MLGPoints,
           doritos.count,dew.count);
    return 0;
}

int CalcPtsTurn(void) {
    int dor = doritos.count*doritos.PtsTurn;
    int bdew = dew.count*dew.PtsTurn;
    return dor+bdew;
}
